common/src/main/java/com/sequenceiq/cloudbreak/converter/DefaultEnumConverter.java

```diff
@@ -24,9 +26,12 @@ public E convertToEntityAttribute(String attribute) {
         } catch (Exception e) {
             LOGGER.info("The {} value is not backward compatible: {}", getDefault().getClass().getSimpleName(), attribute);
         }
-        return getDefault();
+        Optional<E> convertedEnum = tryConvert(attribute);
+        return convertedEnum.orElse(getDefault());
     }
 
     public abstract E getDefault();
 
+    public abstract Optional<E> tryConvert(String attribute);
```

Comment:
why we dont add a default implementation here?

Reply:
Ehm, great idea.